pkg/network/tracer/connection/ebpfless/tcp_processor.go

```diff
@@ -85,24 +114,24 @@ func (t *TCPProcessor) updateSynFlag(conn *network.ConnectionStats, st *connecti
 // updateTcpStats is designed to mirror the stat tracking in the windows driver's handleFlowProtocolTcp
 // https://github.com/DataDog/datadog-windows-filter/blob/d7560d83eb627117521d631a4c05cd654a01987e/ddfilter/flow/flow_tcp.c#L91
 func (t *TCPProcessor) updateTcpStats(conn *network.ConnectionStats, st *connectionState, pktType uint8, tcp *layers.TCP, payloadLen uint16) { //nolint:revive // TODO
-	payloadSeq := tcp.Seq + uint32(payloadLen)
+	nextSeq := calcNextSeq(tcp, payloadLen)
 
 	if pktType == unix.PACKET_OUTGOING {
 		conn.Monotonic.SentPackets++
-		if !st.hasSentPacket || isSeqBefore(st.maxSeqSent, payloadSeq) {
+		if !st.hasSentPacket || isSeqBefore(st.maxSeqSent, nextSeq) {
 			st.hasSentPacket = true
 			conn.Monotonic.SentBytes += uint64(payloadLen)
-			st.maxSeqSent = payloadSeq
+			st.maxSeqSent = nextSeq
 		}
 
 		ackOutdated := !st.hasLocalAck || isSeqBefore(st.lastLocalAck, tcp.Ack)
 		if tcp.ACK && ackOutdated {
 			// wait until data comes in via SynStateAcked
 			if st.hasLocalAck && st.remoteSynState == SynStateAcked {
 				ackDiff := tcp.Ack - st.lastLocalAck
-				// if this is ack'ing a fin packet, there is an extra sequence number to cancel out
-				isFinAck := st.hasRemoteFin && tcp.Ack == st.remoteFinSeq+1
+				isFinAck := st.hasRemoteFin && tcp.Ack == st.remoteFinSeq
```

Comment:
Is this supposed to not have the `+1` at the end?

Reply:
Good eye, I moved this +1 logic into `nextSeq++` in `calcNextSeq` [located here](https://github.com/DataDog/datadog-agent/pull/31382/files#diff-e9f8000b8b7dc0a19fd53cb81219d3dbf079d6576f4434e0b19205308ab8cf1dR68).